lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2322,6 +2323,13 @@ instance LiftHandler ErrStartTimeLaterThanEndTime where
         , "'."
         ]
 
+instance LiftHandler PastHorizonException where
+    handler _ = apiError err400 PastHorizon $ mconcat
+        [ "The end time is past the horizon (due to uncertainty about the next"
```

Comment:
The phrase "The end time" is specific to `ErrListTransactionsPastHorizonException`, whereas _this_ instance could theoretically be thrown from any endpoint, I believe.

How about either:
1. Make the the description generic (remove "The end time")
2. Define this message directly in the `ErrListTransactionsPastHorizonException` case

Reply:
🤔 I think
> Define this message directly in the ErrListTransactionsPastHorizonException case

would be preferable.

If a `PastHorizonException` were to be thrown in many other places, that would be an unexpected server error, different from this case.

Reply:
So, IF `PastHorizonException` always means that a later request will succeed, then we can just have the LiftHandler instance return 503. Is that the case?

Reply:
> always means that a later request will succeed, then we can just have the LiftHandler instance return 503. Is that the case?

There should exist cases where it's more like "programmer error", and waiting doesn't really help.

For instance:

When listing stake pools, we convert the retirement epochs to `UTCTime`:
https://github.com/input-output-hk/cardano-wallet/blob/90c1e5953a48ffc65f004a68abd545ef68574c4b/lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs#L351-L352

In Byron;Shelley;Gougen, before fork to Gougen is confirmed, because the retirements are far in the future, this would likely fail*.

Even if we could wait for the retirement epoch of one pool to enter the forecast range, there could always be additional pools retiring. So waiting won't really help, we need to approach this differently when the time comes.

(*Not a problem currently in Byron;Shelley because pools only exist in Shelley, guaranteeing unlimited forecast)

Reply:
Although "Wait for the node to finish syncing to the hard fork", before the Gougen hard-fork would be clear I suppose